evalml/utils/gen_utils.py

```diff
@@ -267,20 +266,21 @@ def safe_repr(value):
     return repr(value)
 
 
-def is_all_numeric(df):
-    """Checks if the given DataFrame contains only numeric values
+def is_all_numeric(dt):
+    """Checks if the given DataTable contains only numeric values
 
     Arguments:
-        df (DataFrame): The DataFrame to check datatypes of
+        dt (ww.DataTable): The DataTable to check data types of
 
     Returns:
-        True if all the DataFrame columns are numeric and are not missing any values, False otherwise
+        True if all the DataTable columns are numeric and are not missing any values, False otherwise
     """
-    if df.isnull().any().any():
-        return False
-    for dtype in df.dtypes:
-        if dtype not in numeric_and_boolean_dtypes:
+    for ltype in dt.logical_types.values():
```

Comment:
@angela97lin can't we use the "numeric" semantic tag to simplify this, and ignore the logical types?